lib/matplotlib/mathtext.py

```diff
@@ -2873,25 +2873,37 @@ def accent(self, s, loc, toks):
                 ])
 
     def function(self, s, loc, toks):
-        self.push_state()
-        state = self.get_state()
-        state.font = 'rm'
-        hlist = Hlist([Char(c, state) for c in toks[0]])
-        self.pop_state()
+        hlist = self.operatorname(s, loc, toks)
         hlist.function_name = toks[0]
         return hlist
 
     def operatorname(self, s, loc, toks):
         self.push_state()
         state = self.get_state()
         state.font = 'rm'
+        hlist_list = []
         # Change the font of Chars, but leave Kerns alone
         for c in toks[0]:
             if isinstance(c, Char):
                 c.font = 'rm'
                 c._update_metrics()
+                hlist_list.append(c)
+            elif isinstance(c, str):
+                hlist_list.append(Char(c, state))
+            else:
+                hlist_list.append(c)
+        next_char_loc = loc + len(toks[0]) + 1
+        if isinstance(toks[0], ParseResults):
+            next_char_loc += 14  # Skip `operatorname{}`
```

Comment:
I'd just insert it in there:
```suggestion
            next_char_loc += len('operatorname{}')
```